controller/controlplane/manager_options.go

```diff
@@ -133,6 +133,14 @@ func WithFeatureGates(logger logr.Logger, featureGates []gwtypes.ControlPlaneFea
 	}
 }
 
+// WithQPSAndBurst sets the QPS and burst for the API server.
+func WithQPSAndBurst(qps float32, burst int) managercfg.Opt {
+	return func(c *managercfg.Config) {
+		c.APIServerQPS = int(qps)
```

Comment:
Any particular reason to accept a float32 when we need an int?

```suggestion
func WithQPSAndBurst(qps int, burst int) managercfg.Opt {
	return func(c *managercfg.Config) {
		c.APIServerQPS = qps
```

Reply:
because on the caller site we have `float32`, and having it like that prevents from an ordering mistake. But I agree, one way or another it's a little bit ugly at the end of the day

Reply:
Perhaps adding new types QPS and Burst for these would solve this more elegantly and prevent mistakes? WDYT?

Reply:
But still, from CLI we take `int`, but here we need to pass `float32`, and in KIC subsystem we have the same; thus, forth and back conversion is needed. When we get rid of cli portion of KIC config will be able to be adjusted.

Type system in Go still doesn't prevent you from making a mistake. You can pass untyped `const` as whatever type that wraps a numeric type, and it will compile. 

So tbh, from my perspective, it doesn't matter much where conversion happens in this case. I can move it upper if you prefer that way. WDYT?

Reply:
> Type system in Go still doesn't prevent you from making a mistake. You can pass untyped const as whatever type that wraps a numeric type, and it will compile.

Sadly that a reality we have to live in. Adding new types for arguments would at least help in suggesting to user which argument should use which type.

I see your point about using KIC's config fields. I'm OK leaving this as is.